TowerNinja/Assets/Scripts/Fireball.cs

```diff
@@ -104,6 +108,13 @@ private void Die()
         Destroy(gameObject);
     }
 
+    private void GiveManaReward()
+    {
+        GameObject resourceManagerObject = GameObject.Find("ResourceManager").gameObject;
+        ResourceManager resourceManager = resourceManagerObject.GetComponent<ResourceManager>();
```

Comment:
You can simplify the above to
`var resourceManager = FindObjectOfType<ResourceManager>()`